Tests/MLXTests/ModuleTests.swift

```diff
@@ -518,6 +518,52 @@ class ModuleTests: XCTestCase {
         }
     }
 
+    func testLinearUpdateParametersAllSet() throws {
+        class M: Module {
+            let a: Linear
+            let b: Linear
+
+            override init() {
+                self.a = Linear(1, 2)
+                self.b = Linear(3, 4)
+            }
+        }
+        let m = M()
+        let weightsA = MLXArray(0 ..< 2).reshaped([2, 1])
+        let weightsB = MLXArray(0 ..< 12).reshaped([4, 3])
+        let biasA = MLXArray(0 ..< 2)
+        let biasB = MLXArray(0 ..< 4)
+
+        try m.update(
+            parameters: .init(
+                item: .dictionary([
+                    "a": .dictionary(["weight": .value(weightsA), "bias": .value(biasA)]),
+                    "b": .dictionary(["weight": .value(weightsB), "bias": .value(biasB)]),
+                ])),
+            verify: .all)
+        let wA = m.a.weight.asArray(Int.self)
+        XCTAssertEqual(wA, [0, 1])
+        let wB = m.b.weight.asArray(Int.self)
+        XCTAssertEqual(wB, Array(0 ..< 12))
+        XCTAssertThrowsError(
+            try m.update(
+                parameters: .init(
+                    item: .dictionary([
+                        "a": .dictionary(["weight": .value(weightsA)]),
+                        "b": .dictionary(["weight": .value(weightsB)]),
+                    ])),
+                verify: .all)
+        ) { error in
+            guard let error = error as? UpdateError,
+                case let .keyNotFound(base: base, key: key) = error
+            else {
+                XCTFail("Expected to fail with UpdateError.mismatchedSize, but got: \(error)")
```

Comment:
this fail looks like it might be a cut & paste -- the error should be `UpdateError.keyNotFound`

Reply:
You're totally right, this was coming from a cut and paste and hasn't been adapted (though it's only the error when the test fails so it wasn't triggered), should be fixed now.